test-data/run-classes.test

```diff
@@ -57,6 +57,157 @@ assert hasattr(c, 'x')
+
+class C:
+    def __init__(self) -> None:
+        self.c = 3
+
+    def get_c(self) -> int:
+        return self.c
+
+class B(C):
+    def __init__(self) -> None:
+        super().__init__()
+        self.b = 2
+
+    def get_b(self) -> int:
+        return self.b
+
+@decorator
+class A(B):
+    def __init__(self) -> None:
+        super().__init__()
+        self.a = 1
+
+    @classmethod
+    def constant(self) -> int:
+        return 4
+
+    def get_a(self) -> int:
+        return self.a
+
+[file driver.py]
+from native import A
+a = A()
+assert a.a == 1
+assert a.b == 2
+assert a.c == 3
+assert a.get_a() == 1
+assert a.get_b() == 2
+assert a.get_c() == 3
+
+[case testEnum]
+from enum import Enum
+
+class TestEnum(Enum):
+    a : int = 1
+    b : int = 2
+
+    @classmethod
+    def test(cls) -> int:
+        return 3
+
+assert TestEnum.test() == 3
+
+[file driver.py]
+from native import TestEnum
+assert TestEnum.a.name == 'a'
+assert TestEnum.a.value == 1
+assert TestEnum.b.name == 'b'
+assert TestEnum.b.value == 2
+
+[case testDataClass]
+from dataclasses import dataclass, field
+from typing import Set, List, Callable, Any
+
+@dataclass
+class Person1:
+    age : int
+    name : str
+
+@dataclass
+class Person2:
+    age : int
+    name : str = 'robot'
+
+@dataclass
+class Person3:
+    age : int = field(default = 6)
+    friendIDs : List[int] = field(default_factory = list)
+
+    def get_age(self) -> int:
+        return (self.age)
+
+    def set_age(self, new_age : int) -> None:
+        self.age = new_age
+
+    def add_friendID(self, fid : int) -> None:
+        self.friendIDs.append(fid)
+
+    def get_friendIDs(self) -> List[int]:
+        return self.friendIDs
+
+def get_next_age(g: Callable[[Any], int]) -> Callable[[Any], int]:
+    def f(a: Any) -> int:
+        return g(a) + 1 
+    return f
+
+@dataclass
+class Person4:
+    age : int
+    _name : str = 'Bot'
+
+    @get_next_age
+    def get_age(self) -> int:
+        return self.age
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+[file driver.py]
+import sys
+
+# Dataclasses introduced in 3.7
+version = sys.version_info[:2]
+if version[0] < 3 or version[1] < 7:
+    exit()
+
+from native import Person1, Person2, Person3, Person4
+i1 = Person1(age = 5, name = 'robot')
+assert i1.age == 5
+assert i1.name == 'robot'
+i2 = Person2(age = 5)
+assert i2.age == 5
+assert i2.name == 'robot'
+i3 = Person2(age = 5, name = 'new_robot')
+assert i3.age == 5
+assert i3.name == 'new_robot'
+i4 = Person3()
+assert i4.age == 6
+assert i4.friendIDs == []
+i5 = Person3(age = 5)
+assert i5.age == 5
+assert i5.friendIDs == []
+i6 = Person3(age = 5, friendIDs = [1,2,3])
+assert i6.age == 5
+assert i6.friendIDs == [1,2,3]
+assert i6.get_age() == 5
+i6.set_age(10)
+assert i6.get_age() == 10
+i6.add_friendID(4)
+assert i6.get_friendIDs() == [1,2,3,4]
+i7 = Person4(age = 5)
+assert i7.get_age() == 6
+i7.age += 3
+assert i7.age == 8
+assert i7.name == 'Bot'
+
```

Comment:
Just in case I would add a test for generated methods like `__lt__()` etc.

Reply:
Wasn't marked as outdated, but done!